haystack/components/eval/sas_evaluator.py

```diff
@@ -89,42 +69,49 @@ def from_dict(cls, data: Dict[str, Any]) -> "SASEvaluator":
             data["init_parameters"]["device"] = ComponentDevice.from_dict(device)
         return default_from_dict(cls, data)
 
-    @component.output_types(sas=float, scores=List[float])
-    def run(self, predictions: List[str]) -> Dict[str, Any]:
-        if len(predictions) != len(self._labels):
-            raise ValueError("The number of predictions and labels must be the same.")
-
-        if len(predictions) == 0:
-            return {"sas": 0.0, "scores": [0.0]}
-
+    def warm_up(self):
+        """
+        Load the model used for evaluation
+        """
         token = self._token.resolve_value() if self._token else None
-
-        predictions = _preprocess_text(
-            predictions, self._regexes_to_ignore, self._ignore_case, self._ignore_punctuation, self._ignore_numbers
-        )
-        labels = _preprocess_text(
-            self._labels, self._regexes_to_ignore, self._ignore_case, self._ignore_punctuation, self._ignore_numbers
-        )
         config = AutoConfig.from_pretrained(self._model, use_auth_token=token)
         cross_encoder_used = False
         if config.architectures:
             cross_encoder_used = any(arch.endswith("ForSequenceClassification") for arch in config.architectures)
-
-        device = ComponentDevice.resolve_device(self._device)
-
+        device = ComponentDevice.resolve_device(self._device).to_torch_str()
         # Based on the Model string we can load either Bi-Encoders or Cross Encoders.
         # Similarity computation changes for both approaches
-
         if cross_encoder_used:
-            # For Cross Encoders we create a list of pairs of predictions and labels
-            similarity_model = CrossEncoder(
+            self._similarity_model = CrossEncoder(
                 self._model,
-                device=device.to_torch_str(),
+                device=device,
                 tokenizer_args={"use_auth_token": token},
                 automodel_args={"use_auth_token": token},
             )
+        else:
+            self._similarity_model = SentenceTransformer(self._model, device=device, use_auth_token=token)
+
+    @component.output_types(sas=float, scores=List[float])
+    def run(self, labels: List[str], predictions: List[str]) -> Dict[str, Any]:
+        """
+        Run the SASEvaluator to compute the Semantic Answer Similarity (SAS) between a list of predictions and a list of
+        labels. Both must be list of strings of same length.
+        Returns a dictionary containing the SAS score and a list of similarity scores for each prediction-label pair.
```

Comment:
```suggestion
        Run the SASEvaluator to compute the Semantic Answer Similarity (SAS).
        
        :param predictions: List of predictions.
        :param labels: List of labels against which the predictions are compared.
        :returns: A dictionary with the following outputs:
            		     * `sas` - Cumulative SAS score for the entire dataset.
            		     * `scores` - A list of similarity scores for each prediction-label pair.
```
Let's be as explicit as we can.